src/leases/components/leaseSections/summary/LeaseHistory.js

```diff
@@ -0,0 +1,197 @@
+// @flow
+import React, {PureComponent} from 'react';
+import {connect} from 'react-redux';
+
+import LeaseHistoryItem from './LeaseHistoryItem';
+import TitleH3 from '$components/content/TitleH3';
+import {LeaseFieldPaths, LeaseFieldTitles} from '$src/leases/enums';
+import {getContentRelatedLeasesFrom, getContentRelatedLeasesTo, sortRelatedLeasesFrom} from '$src/leases/helpers';
+import {getFieldOptions} from '$src/util/helpers';
+import {getUiDataLeaseKey} from '$src/uiData/helpers';
+import {
+  getAttributes as getLeaseAttributes,
+  getCurrentLease,
+} from '$src/leases/selectors';
+
+import type {Attributes} from '$src/types';
+import type {Lease} from '$src/leases/types';
+import { restructureLease, sortRelatedHistoryItems } from "$src/leases/helpers";
+
+type Props = {
+  currentLease: Lease,
+  leaseAttributes: Attributes,
+}
+
+type State = {
+  currentLease: Lease,
+  leaseAttributes: Attributes,
+  relatedLeasesFrom: Array<Object>,
+  relatedLeasesTo: Array<Object>,
+  stateOptions: Array<Object>,
+}
+
+class LeaseHistory extends PureComponent<Props, State> {
+  state = {
+    currentLease: {},
+    leaseAttributes: null,
+    relatedLeasesFrom: [],
+    relatedLeasesTo: [],
+    stateOptions: [],
+  }
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    const newState = {};
+
+    if(props.leaseAttributes !== state.leaseAttributes) {
+      newState.leaseAttributes = props.leaseAttributes;
+      newState.stateOptions = getFieldOptions(props.leaseAttributes, LeaseFieldPaths.STATE);
+    }
+
+    if(props.currentLease !== state.currentLease) {
+      newState.currentLease = props.currentLease;
+      newState.relatedLeasesFrom = sortRelatedLeasesFrom(getContentRelatedLeasesFrom(props.currentLease));
+      newState.relatedLeasesTo = getContentRelatedLeasesTo(props.currentLease);
+    }
+
+    return newState;
+  }
+
+  render() {
+    const {currentLease} = this.props;
+    const {
+      relatedLeasesFrom,
+      relatedLeasesTo,
+      stateOptions,
+    } = this.state;
+
+    const renderLeaseWithPlotSearchesAndApplications = (lease, active) => {
+      const historyItems = []
+
+      if (lease.plot_searches.length) {
+        lease.plot_searches.forEach((plotSearch) => {
+          historyItems.push({
+            key: `plot-search-${plotSearch.name}-${Math.random().toString()}`,
+            id: plotSearch.id,
+            itemTitle: plotSearch.name,
+            startDate: plotSearch.begin_at,
+            endDate: plotSearch.end_at,
+            plotSearchType: plotSearch.type,
+            plotSearchSubtype: plotSearch.subtype,
+            itemType: "Haku",
```

Comment:
Is this string used internally in the app only and not shown to the users? If shown, would it need to be translated?